go/libraries/doltcore/remotestorage/chunk_store.go

```diff
@@ -383,16 +406,24 @@ func (r *locationRefresh) Add(resp *remotesapi.DownloadLoc) {
 	}
 }
 
-func (r *locationRefresh) GetURL(ctx context.Context, client remotesapi.ChunkStoreServiceClient) (string, error) {
+var refreshTableFileURLRetryDuration = 5 * time.Second
+
+func (r *locationRefresh) GetURL(ctx context.Context, lastError error, client remotesapi.ChunkStoreServiceClient) (string, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	if r.RefreshRequest != nil && time.Now().After(r.RefreshAfter) {
-		resp, err := client.RefreshTableFileUrl(ctx, r.RefreshRequest)
-		if err != nil {
-			return r.URL, err
+	if r.RefreshRequest != nil {
+		now := time.Now()
+		wantsRefresh := now.After(r.RefreshAfter) || errors.Is(lastError, HttpError)
+		canRefresh := now.After(r.lastRefresh.Add(refreshTableFileURLRetryDuration))
```

Comment:
think this is easier to read:

canRefresh := time.Since(r.lastRefresh) > refreshTableFileURLRetryDuration